components/tidb_query_expr/src/impl_string.rs

```diff
@@ -32,6 +32,44 @@ pub fn oct_int(num: &Int, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write(Some(Bytes::from(format!("{:o}", num)))))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn oct_string(s: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    if s.is_empty() {
+        return Ok(writer.write(None));
+    }
+
+    let mut trimmed = s.iter().skip_while(|x| x.is_ascii_whitespace());
+    let mut r = Some(0u64);
+    let mut negative = false;
+    let mut overflow = false;
+    if let Some(&c) = trimmed.next() {
```

Comment:
May be we can use https://doc.rust-lang.org/std/primitive.i64.html#method.from_str_radix here？

Reply:
I think we need to handle more flexible input than `from_str_radix`, eg: `select oct('12a');`  the result should be `14`, and some other overflow-related issues, like follow test cases:
https://github.com/tikv/tikv/blob/2665aea33e1b30eb261bdbb86bcffa7c08207986/components/tidb_query_expr/src/impl_string.rs#L1026-L1037
`from_str_radix` maybe not much helpful. 
This is a 2019's PR, I almost forgot why I wrote this way. Feel free to give more context if I am missing something 😄

Reply:
Well, it looks like TiDB is not compatible with MySQL here:

https://github.com/pingcap/tidb/blob/094b462b445121ab8a4f6ef3578d9d65b283e911/expression/builtin_string.go#L2798-L2805

Maybe we to do following:

- Submit a issue and add it to https://github.com/pingcap/tidb/issues/11223
- Until TiDB fix this, we should keep the logic works the same with TiDB on TiKV side.

cc @zhongzc any ideas?

Reply:
Try `i128:: from_str_radix(s.split(|c| c < '0' || c > '8').first().unwrap_or(""), 8)`

Reply:
Is there an equivalent to POSIX `strtoll` in Rust?

Reply:
> Try `s.split(|c| c < '0' || c > '8').first().and_then(|digits| i128::from_str_radix(digits, 8).ok())`?

Not like that, the input string is base on 10.

Reply:
OK